libs/x-content/src/test/java/org/elasticsearch/common/xcontent/ObjectParserTests.java

```diff
@@ -1112,4 +1114,31 @@ public void testFieldsForVersionsOnOrAfter() throws IOException {
         StructWithOnOrAfterField o2 = StructWithOnOrAfterField.PARSER.parse(futureParser, null);
         assertEquals(1, o2.intField);
     }
+
+    public static class DoubleFieldDeclaration {
+        static final ObjectParser<DoubleFieldDeclaration, Void> PARSER =
+            new ObjectParser<>("double_field_declaration", DoubleFieldDeclaration::new);
+        static {
+
+            PARSER.declareInt(DoubleFieldDeclaration::setIntField, new ParseField("name"));
+            PARSER.declareInt(DoubleFieldDeclaration::setIntField, new ParseField("name"));
+
+        }
+
+        private int intField;
+
+        private  void setIntField(int intField) {
+            this.intField = intField;
+        }
+    }
+
+    public void testDoubleDeclarationThrowsException() throws IOException {
+        XContentParser parser = createParser(JsonXContent.jsonXContent, "{\"name\": 1}");
+
+        ExceptionInInitializerError error = expectThrows(ExceptionInInitializerError.class,
+            () -> DoubleFieldDeclaration.PARSER.parse(parser, null));
```

Comment:
Could you move the parser declaration into the test method and assert that we throw on the second `declareInt`? I think its important that we throw there instead of in the `parse` call. I think what you built already does that and this line still works because the PARSER is initialized on it.

Reply:
agree, the direct use of parser is more clear here